org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/EclipseWorkspaceArtifactRepository.java

```diff
@@ -77,8 +77,8 @@ protected File resolveAsEclipseProject(String groupId, String artifactId, String
 
     if(context.resolverConfiguration.isResolveWorkspaceProjects()) {
       IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
-      IPath file = pom.getLocation();
-      if(file == null) {
+      IPath pomFile = pom.getLocation();
+      if(pomFile == null) {
         return ProjectRegistryManager.toJavaIoFile(pom);
       }
       if(!POM_EXTENSION.equals(extension)) {
```

Comment:
this needs further cleanup as well, as actually POM_EXTENSION is the type (not the extension), but the type is only passed from the legacy https://github.com/eclipse-m2e/m2e-core/pull/1731/files#diff-22a899eac2926341327db3736d6ed24c7f9ce9a64a154f805af15b424d6fed75L146 but not from the modern resolver SPI https://github.com/eclipse-m2e/m2e-core/pull/1731/files#diff-22a899eac2926341327db3736d6ed24c7f9ce9a64a154f805af15b424d6fed75R141

Reply:
Seems that `pom` type is having extension `pom` (https://maven.apache.org/ref/3.9.6/maven-core/artifact-handlers.html) despite its file name `pom.xml`.

Reply:
If you look at how the pom files are published in Maven repos (eg https://repo.eclipse.org/content/repositories/tycho/org/eclipse/tycho/tycho-core/4.0.8-SNAPSHOT/ ), then you can see that they are indeed published as `artifactId-version.pom` . The `xml` is the "source" extension, but the derived artifact is a `.pom` file.